notification-server/src/google_cloud.rs

```diff
@@ -357,17 +363,213 @@ mod tests {
             server.url(),
             "/v1/projects/fake_project_id/messages:send".to_string()
         );
-        let notification_information = NotificationInformation::new(
-            AccountAddress::from_str("4FmiTW2L2AccyR9VjzsnpWFSAcohXWf7Vf797i36y526mqiEcp").unwrap(),
-            "100".to_string(),
-            Preference::CCDTransaction,
+        let notification_information =
+            NotificationInformation::CCD(CCDTransactionNotificationInformation::new(
+                AccountAddress::from_str("4FmiTW2L2AccyR9VjzsnpWFSAcohXWf7Vf797i36y526mqiEcp")
+                    .unwrap(),
+                "100".to_string(),
+            ));
+        assert!(gc
+            .send_push_notification("valid_device_token", &notification_information)
+            .await
+            .is_ok());
+        mock.assert();
+    }
+
+    #[tokio::test]
+    async fn test_send_push_notification_cis2_without_metadata_url() {
+        let mut server = mockito::Server::new_async().await;
+        let mock_provider = MockTokenProvider {
+            token_response: Arc::new("mock_token".to_string()),
+            should_fail:    false,
+        };
+
+        let expected_body = json!({
+            "message": {
+                "token": "test_token",
+                "data": {
+                    "recipient": "3kBx2h5Y2veb4hZgAJWPrr8RyQESKm5TjzF3ti1QQ4VSYLwK1G",
+                    "amount": "200",
+                    "type": "cis2-tx",
+                    "token_id": "ffffff",
+                    "contract_index": "111",
```

Comment:
@Radiokot @lassemand 
Would make sense to use the full contract address here as contracts could potentially be identified by a subindex as well (not relevant now, but could be in the future). Adding it now would avoid introducing breaking changes when you do need it.

Also, contract addresses are usually serialized as `{"index": <u64>, "subindex": <u64>}`. Any reason we don't just stick to the standard?

If you wanted to be even more "standard conforming" there is a [`TokenAddress`](https://proposals.concordium.software/CIS/cis-2.html#token-address) type in the CIS2 standard that uniquely identifies CIS2 tokens (consists of contract address + token ID).

Reply:
1. Adding an extra `contract_subindex` field in the future won't be a breaking change, as well as adding it now is not hard. Let's have it now;
2. I didn't know it is a standard. If it makes sense, let's change `contract_index` to `contract_address` with the corresponding JSON.

Reply:
I am sticking with the definition defined in https://concordium.atlassian.net/wiki/spaces/EN/pages/1502019590/Message+data+payload

We discussed it and they are currently just throwing away the subindex. 

I will let @Radiokot decide

Reply:
I think it would be bad to throw away the subindex? A contract address as defined by the GRPC API includes an index/subindex pair so if the subindex is discarded you'll just have to recreate it under the assumption that it was 0 which i think is a little pointless and on top of that possibly wrong (in the future) 🤷